params/astria_config.go

```diff
@@ -0,0 +1,369 @@
+		BridgeAddresses:     make(map[string]*AstriaBridgeAddressConfig),
+		BridgeAllowedAssets: make(map[string]struct{}),
+	}
+}
+
+func (c *AstriaForks) GetForkAtHeight(height uint64) AstriaForkData {
+	idx := sort.Search(len(c.orderedForks), func(i int) bool {
+		return c.orderedForks[i].Height <= height
+	})
+	// no named fork at this height
+	if idx == len(c.orderedForks) {
+		return GetDefaultAstriaForkData()
+	}
+	return c.orderedForks[idx]
+}
+
+func (c *AstriaForks) GetNextForkAtHeight(height uint64) *AstriaForkData {
+	idx := sort.Search(len(c.orderedForks), func(i int) bool {
+		return c.orderedForks[i].Height > height
+	})
+	if idx == len(c.orderedForks) {
+		return nil
+	}
+	return &c.orderedForks[idx]
+}
+
+func (c *AstriaForks) MinBaseFeeAt(height uint64) *big.Int {
+	return big.NewInt(0).SetUint64(c.GetForkAtHeight(height).EIP1559Params.MinBaseFee)
+}
+
+// BaseFeeChangeDenominator bounds the amount the base fee can change between blocks.
+func (c *AstriaForks) BaseFeeChangeDenominatorAt(height uint64) uint64 {
+	return c.GetForkAtHeight(height).EIP1559Params.BaseFeeChangeDenominator
+}
+
+// ElasticityMultiplier bounds the maximum gas limit an EIP-1559 block may have.
+func (c *AstriaForks) ElasticityMultiplierAt(height uint64) uint64 {
+	return c.GetForkAtHeight(height).EIP1559Params.ElasticityMultiplier
+}
+
+func (c *AstriaForks) MarshalJSON() ([]byte, error) {
+	return json.Marshal(c.forkMap)
+}
+
+func (c *AstriaForks) UnmarshalJSON(data []byte) error {
+	var forkMap map[string]AstriaForkConfig
+	if err := json.Unmarshal(data, &forkMap); err != nil {
+		return err
+	}
+
+	newForks, err := NewAstriaForks(forkMap)
+	if err != nil {
+		return err
+	}
+
+	*c = *newForks
+	return nil
+}
+
+func (c *ChainConfig) GetAstriaForks() *AstriaForks {
+	if c.AstriaForks == nil {
+		forks, _ := NewAstriaForks(nil)
+		return forks
+	}
+	return c.AstriaForks
+}
+
+type AstriaBridgeAddressConfig struct {
+	BridgeAddress  string                  `json:"bridgeAddress"`
+	SenderAddress  common.Address          `json:"senderAddress,omitempty"`
+	StartHeight    uint32                  `json:"startHeight"`
+	AssetDenom     string                  `json:"assetDenom"`
+	AssetPrecision uint16                  `json:"assetPrecision"`
+	Erc20Asset     *AstriaErc20AssetConfig `json:"erc20Asset,omitempty"`
+}
+
+type AstriaErc20AssetConfig struct {
+	ContractAddress   common.Address `json:"contractAddress"`
+	ContractPrecision uint16         `json:"contractPrecision"`
+}
+
+func (abc *AstriaBridgeAddressConfig) Validate(genesisPrefix string) error {
+	prefix, byteAddress, err := bech32.Decode(abc.BridgeAddress)
+	if err != nil {
+		return fmt.Errorf("bridge address must be a bech32 encoded string")
+	}
+	byteAddress, err = bech32.ConvertBits(byteAddress, 5, 8, false)
+	if err != nil {
+		return fmt.Errorf("failed to convert address to 8 bit")
+	}
+	if prefix != genesisPrefix {
+		return fmt.Errorf("bridge address must have prefix %s", genesisPrefix)
+	}
```

Comment:
it has to have the same prefix? what if the sequencer network was migrated and the prefixes are different?

Reply:
the address prefix is part of the `AstriaSequencerConfig` so can be changed in the genesis config if sequencer network changes. I think `genesisPrefix` is just leftover from the previous validation checks but the var name is confusing here. will update.